### ISSUE TITLE
[Bug] ModelListGP.fantasize raises unexpected exception

### ISSUE THREAD

**Issue by JackBuck** [pytorch/botorch#1265]
# 🐛 Bug

The `fantasize` function for the `ModelListGP` raises an exception when I believe it shouldn't. I think that this is because the `transform_inputs` function on the `ModelList` class (a super class of `ModelListGP`) returns a list of `Tensor`s instead of a single `Tensor`.

## To reproduce

**Code snippet to reproduce**
```python
import torch
from botorch.models import FixedNoiseGP, ModelListGP
from botorch.sampling import SobolQMCNormalSampler

train_x = torch.arange(12, dtype=torch.float).reshape(6, 2)
train_obj = torch.ones(6, 2, dtype=torch.float)
train_var = torch.zeros(6, 2, dtype=torch.float)

model = ModelListGP(
    FixedNoiseGP(train_x, train_obj[..., :1], train_var[..., :1]),
    FixedNoiseGP(train_x, train_obj[..., 1:2], train_var[..., 1:2]),
)

model.fantasize(
    X=torch.arange(10, dtype=torch.float).reshape(5, 1, 2),
    sampler=SobolQMCNormalSampler(16),
)
```

**Stack trace/error message**
```
Traceback (most recent call last):
  File ".../mwe_modellist.py", line 15, in <module>
    model.fantasize(
  File ".../python3.10/site-packages/botorch/models/model.py", line 168, in fantasize
    return self.condition_on_observations(
  File ".../python3.10/site-packages/botorch/models/model_list_gp_regression.py", line 93, in condition_on_observations
    return super().get_fantasy_model(inputs, targets, **kwargs_)
  File ".../python3.10/site-packages/gpytorch/models/model_list.py", line 74, in get_fantasy_model
    fantasy_models = [
  File ".../python3.10/site-packages/gpytorch/models/model_list.py", line 75, in <listcomp>
    model.get_fantasy_model(*inputs_, *targets_, **kwargs_)
TypeError: ExactGP.get_fantasy_model() takes 3 positional arguments but 4 were given
```

## Expected Behavior

I would expect this code to not raise an exception. It should return a `ModelListGP` with the additional (simulated/fantasized) training data.

## System information

- BoTorch Version: 0.6.4
- GPyTorch Version: 1.6.0
- PyTorch Version: 1.11.0
- Python Version: 3.10.5
- Computer OS: Ubuntu 20.04.4 LTS (Focal Fossa)

## Additional context

As mentioned above, I think that the problem is an inconsistency in the return type of `transform_inputs`.
  - `Model.fantasize` calls `self.transform_inputs(X)` and passes this to `self.condition_on_observations`
  - `ModelListGP.transform_inputs` returns a list of tensors (implementation inherited from `ModelList`)
  - `ModelListGP.condition_on_observations` expects the `X` input to be a `Tensor`, not a list of tensors

**Comment by saitcakmak** [comment]
Hi @JackBuck! Thanks for flagging the issue. I believe this was fixed by #1250. It should work if you install latest botorch via pip install --upgrade git+https://github.com/pytorch/botorch.git

**Comment by JackBuck** [comment]
Amazing, fixed in negative time! Thanks @saitcakmak .
